backend/main/test/Food/Domain/Food.test.ts

```diff
@@ -0,0 +1,140 @@
+import { describe, expect, it, test } from "vitest";
+import { Food } from "../../../src/Group/Domain/Entities/Food";
+import { FoodDomainError } from "../../../src/Group/Domain/Entities/Food";
+import { Quantity } from "../../../src/Group/Domain/ValueObjects/Quantity";
+import { Expiry } from "../../../src/Group/Domain/ValueObjects/Expiry";
+import { Unit } from "../../../src/Group/Domain/ValueObjects/Unit";
+
+describe("Food Object", () => {
+	describe("creating food object", () => {
+		const unit = Unit.create({ name: "g" });
+		const quantity = Quantity.create(100);
+		const expiry = Expiry.create({ date: new Date(2023, 11, 1) });
+		const foodId = "foodId";
+		const fullFoodProps = {
+			name: "dummy food name",
+			unit: unit,
+			quantity: quantity,
+			expiry: expiry,
+		};
+
+		const requiredFoodProps = {
+			name: "dummy food name",
+		};
+
+		it("success with full food props", () => {
+			const food = Food.create(foodId, {
+				...fullFoodProps,
+			});
+			expect(food.ok).toBeTruthy();
+			expect(food.value!.name).toBe(fullFoodProps.name);
+		});
+
+		it("success with required food props", () => {
+			const food = Food.create(foodId, {
+				...requiredFoodProps,
+			});
+			expect(food.ok).toBeTruthy();
+			expect(food.value!.name).toBe(fullFoodProps.name);
+			expect(food.value!.unit).toBeUndefined();
+			expect(food.value!.quantity).toBeUndefined();
+			expect(food.value!.expiry).toBeUndefined();
+		});
+
+		it("food name too long", () => {
+			const food = Food.create(foodId, {
+				name: "abcdefghijklmnopqlstuvwxyzabcdefghijklmnopqlstuvwxy", // 51 characters
+			});
+			expect(food.ok).toBeFalsy();
+		});
+	});
+});
+
+describe("Food Object", () => {
+	const unit = Unit.create({ name: "g" }).value;
+	const expiry = Expiry.create({ date: new Date(2023, 11, 1) }).value;
+	const foodId = "foodId";
+
```

Comment:
This is not wrong but I think it can create food value here. If you did so, you no longer need to create food objects in this 'describe' section. It will make your life easier.

To prevent side effects, you can use the following functions.
https://vitest.dev/api/#beforeeach
https://vitest.dev/api/#aftereach